common/authentication/postgresql/metadata.go

```diff
@@ -86,16 +86,30 @@ func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts Ini
 	return nil
 }
 
-func (m *PostgresAuthMetadata) ValidateAwsIamFields() (string, string, string, error) {
+func (m *PostgresAuthMetadata) GetAWSIAMAuthOptions() (*aws.AWSIAMAuthOptions, error) {
 	awsRegion, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSRegion")
 	if awsRegion == "" {
-		return "", "", "", errors.New("metadata property AWSRegion is missing")
+		return nil, errors.New("metadata property AWSRegion is missing")
 	}
-	// Note: access key and secret keys can be optional
-	// in the event users are leveraging the credential files for an access token.
+	// only region is mandatory
+
 	awsAccessKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
 	awsSecretKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
-	return awsRegion, awsAccessKey, awsSecretKey, nil
+	awsSessionToken, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSessionToken")
+
+	awsIamRoleArn, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "awsIamRoleArn")
+	awsStsSessionName, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "awsStsSessionName")
```

Comment:
```suggestion
	sessionName, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "sessionName")
```

I don't think we need to expose to end users the underlying mechanisms leveraging Sts through which one can assume a new role with new creds

Reply:
this cannot be changed, it must match with metadata.yaml , metadata.yaml has been set to match the existing metadata names for the kafka component, so we have consistency